applications/GeoMechanicsApplication/custom_elements/transient_thermal_element.h

```diff
@@ -0,0 +1,303 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Mohamed Nabi
+//                   John van Esch
+//                   Gennady Markelov
+//
+
+#pragma once
+
+#include "includes/element.h"
+#include "geo_mechanics_application_variables.h"
+#include "custom_constitutive/thermal_dispersion_law.h"
+#include "includes/serializer.h"
+
+namespace Kratos {
+
+template <unsigned int TDim, unsigned int TNumNodes>
+class KRATOS_API(GEO_MECHANICS_APPLICATION) TransientThermalElement : public Element {
+public:
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(TransientThermalElement);
+
+    explicit TransientThermalElement(IndexType NewId = 0) : Element(NewId) {}
+
+    TransientThermalElement(IndexType             NewId,
+                            GeometryType::Pointer pGeometry)
+        : Element(NewId, pGeometry)
+    {
+    }
+
+    TransientThermalElement(IndexType               NewId,
+                            GeometryType::Pointer   pGeometry,
+                            PropertiesType::Pointer pProperties)
+        : Element(NewId, pGeometry, pProperties)
+    {
+    }
+
+    ~TransientThermalElement() override = default;
+    TransientThermalElement(const TransientThermalElement&) = delete;
+    TransientThermalElement& operator=(const TransientThermalElement&) = delete;
+
+    Element::Pointer Create(IndexType               NewId,
+                            const NodesArrayType&   rThisNodes,
+                            PropertiesType::Pointer pProperties) const override
+    {
+        return make_intrusive<TransientThermalElement>(NewId, GetGeometry().Create(rThisNodes), pProperties);
+    }
+
+    Element::Pointer Create(IndexType               NewId,
+                            GeometryType::Pointer   pGeom,
+                            PropertiesType::Pointer pProperties) const override
+    {
+        return make_intrusive<TransientThermalElement>(NewId, pGeom, pProperties);
+    }
+
+    int Check(const ProcessInfo&) const override
+    {
+        KRATOS_TRY
+
+        CheckDomainSize();
+        CheckHasSolutionStepsDataFor(TEMPERATURE);
+        CheckHasSolutionStepsDataFor(DT_TEMPERATURE);
+        CheckHasDofsFor(TEMPERATURE);
+        CheckProperties();
+        CheckForNonZeroZCoordinateIn2D();
+
+        KRATOS_CATCH("")
+
+        return 0;
+    }
```

Comment:
In my opinion it would be nicer if the functions that are called here are placed directly under the check function